File: 2343/2343.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var queryKthSmallestTrimmedNumber = function(root) {
    if (!root) return 0;
    return 1 + Math.max(queryKthSmallestTrimmedNumber(root.left), queryKthSmallestTrimmedNumber(root.right));
};

File: 2343/2343.cs
public class Solution {
    public int QueryKthSmallestTrimmedNumber(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(QueryKthSmallestTrimmedNumber(root.left), QueryKthSmallestTrimmedNumber(root.right));
    }
}